### PR TITLE
Add deno.jsonc file and use --config flag instead of --import-map

### PR DESCRIPTION
**Pull request by selfcontained**
## Summary
This is the first change to offer a bit more flexibility around specifying import maps, and adopt some of the upcoming conventions of deno supporting a config file if present.

Currently to support an `import-map` we have to add the `import-map` flag to all deno commands in our hooks. This means we also require the specified `import_map.json` file to be present, otherwise deno will throw an error. 

By adding a `deno.jsonc` file that specifies the `importMap`, it will allow developers to optionally use one, or not if they prefer not to, and not put us in a spot where we have to have that file present for things to work properly. Currently, it does mean we have to set the `--config=deno.json` flag, which means that `deno.jsonc` file has to be present, but in an upcoming release of deno, it should pick up that file if present without any flag needed, so this sets us up for that.

### CODE REVIEW

File: .slack/slack.json [hunk 1]
```diff
@@ -1,17 +1,17 @@
 {
   "manifest": {
     "script": {
-      "default": "deno run -q --unstable --import-map=import_map.json --allow-read --allow-net https://deno.land/x/deno_slack_builder@0.0.7/mod.ts --manifest"
+      "default": "deno run -q --unstable --config=deno.jsonc --allow-read --allow-net https://deno.land/x/deno_slack_builder@0.0.7/mod.ts --manifest"
```
Comment (selfcontained) [inline .slack/slack.json:4]: in a future release of deno, we should be able to drop this `--config` flag entirely

File: .vscode/settings.json [hunk 2]
```diff
@@ -2,7 +2,6 @@
   "deno.enable": true,
   "deno.lint": true,
   "deno.unstable": true,
-  "deno.importMap": "import_map.json",
```
Comment (selfcontained) [inline .vscode/settings.json:5]: The Deno VS code plugin detects the `deno.jsonc` file automatically, which then detects the import map, so we don't need this anymore.

**Review by shapirone — APPROVED** [review]
LGTM 🥳 

Super simple and means that if we want to build the config we don't need to add any other args 🙌

**Review by mwbrooks — APPROVED** [review]
❤️ Love it! I'm happy to see us starting to adopt `deno.jsonc`. I wasn't able to find a way to directly reference modules in the file, but I'm hoping it's an upcoming feature as well.

**Comment by filmaj** [comment]
Neato!

**Pull request merged**